protocol/nats_jetstream/v3/options.go

```diff
@@ -0,0 +1,123 @@
+/*
+ Copyright 2024 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package nats_jetstream
+
+import (
+	"errors"
+
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nats.go/jetstream"
+)
+
+var (
+	ErrNoConnection              = errors.New("URL or nats connection must be given")
+	ErrNoFilterSubjects          = errors.New("no filter subjects were given")
+	ErrMoreThanOneStream         = errors.New("more than one stream for given filter subjects")
+	ErrNoConsumerConfig          = errors.New("no consumer config was given")
+	ErrMoreThanOneConsumerConfig = errors.New("more than one consumer config given")
+)
+
+// ProtocolOption is the function signature required to be considered an nats.ProtocolOption.
+type ProtocolOption func(*Protocol) error
+
+// WithURL configures the Sender and/or Receiver
+func WithURL(url string, natsOpts ...nats.Option) ProtocolOption {
+	return func(p *Protocol) error {
+		opts := []nats.Option{}
+		opts = append(opts, natsOpts...)
+		p.url = url
+		p.natsOpts = opts
+		p.connOwned = true
+		if p.sender != nil {
+			p.sender.url = url
+			p.sender.natsOpts = opts
+			p.sender.connOwned = true
+		}
+		if p.receiver != nil {
+			p.receiver.url = url
+			p.receiver.natsOpts = opts
+			p.receiver.connOwned = true
+		}
+		return nil
+	}
+}
+
+// WithConnection configures the Sender and/or Receiver
```

Comment:
Please update all option comments with a clear description of what they do e.g., 


```suggestion
// WithConnection uses the provided connection in the protocol sender and receiver
```